homeassistant/components/pooldose/sensor.py

```diff
@@ -0,0 +1,219 @@
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        # has no unit
+    ),
+    PooldoseSensorEntityDescription(
+        key="ofa_orp_value",
+        translation_key="ofa_orp_value",
+        device_class=SensorDeviceClass.DURATION,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        native_unit_of_measurement="min",
+    ),
+    PooldoseSensorEntityDescription(
+        key="ph_calibration_type",
+        translation_key="ph_calibration_type",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        # has no unit
+    ),
+    PooldoseSensorEntityDescription(
+        key="ph_calibration_offset",
+        translation_key="ph_calibration_offset",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        device_class=SensorDeviceClass.VOLTAGE,
+        suggested_display_precision=2,
+        entity_registry_enabled_default=False,
+        native_unit_of_measurement="mV",
+    ),
+    PooldoseSensorEntityDescription(
+        key="ph_calibration_slope",
+        translation_key="ph_calibration_slope",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        device_class=SensorDeviceClass.VOLTAGE,
+        suggested_display_precision=2,
+        entity_registry_enabled_default=False,
+        native_unit_of_measurement="mV",
+    ),
+    PooldoseSensorEntityDescription(
+        key="orp_calibration_type",
+        translation_key="orp_calibration_type",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        # has no unit
+    ),
+    PooldoseSensorEntityDescription(
+        key="orp_calibration_offset",
+        translation_key="orp_calibration_offset",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        device_class=SensorDeviceClass.VOLTAGE,
+        suggested_display_precision=2,
+        entity_registry_enabled_default=False,
+        native_unit_of_measurement="mV",
+    ),
+    PooldoseSensorEntityDescription(
+        key="orp_calibration_slope",
+        translation_key="orp_calibration_slope",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        device_class=SensorDeviceClass.VOLTAGE,
+        suggested_display_precision=2,
+        entity_registry_enabled_default=False,
+        native_unit_of_measurement="mV",
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: PooldoseConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up PoolDose sensor entities from a config entry."""
+    coordinator = config_entry.runtime_data.coordinator
+    client = config_entry.runtime_data.client
+    serial_number = config_entry.unique_id
+    device_properties = config_entry.runtime_data.device_properties
+
+    # Ensure serial_number is not None
+    if not serial_number:
+        raise ValueError("Config entry must have a unique_id set")
```

Comment:
I would assume we have a serial number at this point, so let's just use an `if TYPE_CHECKING: assert config_entry.unique_id is not None`